src/lwc/geFormRenderer/geFormRenderer.js

```diff
@@ -2080,18 +2171,39 @@ export default class GeFormRenderer extends LightningElement{
     }
 
     updateFormStateWithSuccessfulPurchaseCall(responseBody) {
-        this.updateFormState({
-            [apiNameFor(PAYMENT_ELEVATE_ID)]: responseBody.id,
-            [apiNameFor(PAYMENT_STATUS)]: responseBody.status,
-            [apiNameFor(PAYMENT_CARD_NETWORK)]: responseBody.cardData.brand,
-            [apiNameFor(PAYMENT_LAST_4)]: responseBody.cardData.last4,
-            [apiNameFor(PAYMENT_EXPIRATION_MONTH)]: responseBody.cardData.expirationMonth,
-            [apiNameFor(PAYMENT_EXPIRATION_YEAR)]: responseBody.cardData.expirationYear,
-            [apiNameFor(PAYMENT_DECLINED_REASON)]: '',
-            [apiNameFor(PAYMENT_GATEWAY_ID)]: responseBody.gatewayId,
-            [apiNameFor(PAYMENT_TRANSACTION_ID)]: responseBody.gatewayTransactionId,
-            [apiNameFor(PAYMENT_AUTHORIZED_AT)]: responseBody.authorizedAt,
-        });
+        if (this.isCreditCardTransaction()) {
+            this.updateFormState({
+                [apiNameFor(PAYMENT_ELEVATE_ID)]: responseBody.id,
+                [apiNameFor(PAYMENT_STATUS)]: responseBody.status,
+                [apiNameFor(PAYMENT_CARD_NETWORK)]: responseBody.cardData.brand,
+                [apiNameFor(PAYMENT_LAST_4)]: responseBody.cardData.last4,
+                [apiNameFor(PAYMENT_EXPIRATION_MONTH)]: responseBody.cardData.expirationMonth,
+                [apiNameFor(PAYMENT_EXPIRATION_YEAR)]: responseBody.cardData.expirationYear,
+                [apiNameFor(PAYMENT_DECLINED_REASON)]: '',
+                [apiNameFor(PAYMENT_GATEWAY_ID)]: responseBody.gatewayId,
+                [apiNameFor(PAYMENT_TRANSACTION_ID)]: responseBody.gatewayTransactionId,
+                [apiNameFor(PAYMENT_AUTHORIZED_AT)]: responseBody.authorizedAt,
+            });
```

Comment:
Do you think it'd be worth extracting this out? Maybe something like `updateFormStateWithCreditCardTransactionResponse(responseBody)`